keps/sig-apps/4563-evacuation-api/README.md

```diff
@@ -0,0 +1,1205 @@
+	// This field is required and immutable.
+	AcceptDeadlineSeconds int32 `json:"acceptDeadlineSeconds" protobuf:"varint,2,opt,name=acceptDeadlineSeconds"`
+	
+	// ProgressDeadlineSeconds is a maximum amount of time an evacuator should take to report on a
+	// progress by updating the .status.evacuationProgressTimestamp.
+	// If the .status.evacuationProgressTimestamp is not updated within the duration of
+	// .spec.progressDeadlineSeconds, the evacuated pod will be evicted using the Eviction API.
+	//
+	// The minimum value is 60 (1m) and the maximum value is 43200 (12h).
+	// The value must be greater than or equal to .spec.acceptDeadlineSeconds.
+	// This field is required and immutable.
+	ProgressDeadlineSeconds int32 `json:"progressDeadlineSeconds" protobuf:"varint,3,opt,name=progressDeadlineSeconds"`
+}
+
+// LocalPodReference contains enough information to locate the referenced pod inside the same namespace.
+type LocalPodReference struct {
+	// Name of the pod.
+	// This field is required.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// UID of the pod.
+	// This field is required.
+	UID string `json:"uid" protobuf:"bytes,2,opt,name=uid"`
+}
+
+// EvacuationStatus represents the most recently observed status of the evacuation.
+type EvacuationStatus struct {
+	// EvacuatorRef references the evacuator of the current evacuation. 
+	// This field should be set when Accepted field is set.
+	// +optional
+	EvacuatorRef *EvacuatorReference `json:"evacuatorRef,omitempty" protobuf:"bytes,1,opt,name=evacuatorRef"`
+
+	// If set to true, EvacuationProgressTimestamp should also be set, and then updated periodically.
+	// If set to false, the pod will be evicted immediately.
+	// This field cannot be changed once set.
+	// +optional
+	Accepted *bool `json:"accepted,omitempty" protobuf:"varint,2,opt,name=accepted"`
+
+	// EvacuationProgressTimestamp is the time at which the evacuation was reported to be in progress by the evacuator.
+	// Cannot be set to the future time (after taking time skew into account).
+	// This field must be set when the Accepted field is set to true.
+	// +optional
+	EvacuationProgressTimestamp *metav1.Time `json:"evacuationProgressTimestamp,omitempty" protobuf:"bytes,3,opt,name=evacuationProgressTimestamp"`
+
+	// ExpectedEvacuationFinishTime is the time at which the evacuation is expected to end.
+	// May be empty if no estimate can be made.
+	// +optional
+	ExpectedEvacuationFinishTime *metav1.Time `json:"expectedEvacuationFinishTime,omitempty" protobuf:"bytes,4,opt,name=expectedEvacuationFinishTime"`
+	
+	// The number of unsuccessful attempts to evict the referenced pod, e.g. due to a PDB.
+	// This field is required.
+	FailedEvictionCounter int32 `json:"failedEvictionCounter" protobuf:"varint,5,opt,name=failedEvictionCounter"`
+	
+	// Message is a human readable message indicating details about the evacuation.
+	// This may be an empty string.
```

Comment:
I am using the same approach as [meta.Condition](https://github.com/kubernetes/kubernetes/blob/9791f0d1f39f3f1e0796add7833c1059325d5098/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1577) does. It can be an empty string, but the field usually should be set and will be present in the transport. We can revisit this in the API review.